netket/experimental/sampler/metropolis_pt.py

```diff
@@ -161,13 +175,39 @@ def __init__(self, *args, n_replicas: int = 32, **kwargs):
             machine_pow: The power to which the machine should be exponentiated to generate the pdf (default = 2).
             dtype: The dtype of the states sampled (default = np.float32).
         """
+        if betas is None:
```

Comment:
my proposal is to not store the betas sorted and all unless the user specifies them. If possible, let's generate them on the fly.

So you have two internal fields, and you should populate the one that the user specified/makes sense.